File: parked/line.js
function line( x1, y1, x2, y2 ) {
    var dx = x2 - x1;
    var dy = y2 - y1;

    if( Math.abs( dx ) > Math.abs( dy ) ) { // x-wise
    }
    else {  // y-wise
        if( y1 > y2 ) { // swap points
            var x = x2; x2 = x1; x1 = x;
            var y = y2; y2 = y1; y1 = y;
        }

        for( var y = y1; y <= y2; y++ ) {
            var x = x1 + Math.round( dx * (y - y1) / dy );
            console.log( x, y );
        }

        console.log( "--" );
        for( var i = 0; i <= dy; i ++ ) {
            var y = y1 + i;
            var x = x1 + Math.round( dx * i / dy );
            console.log( x, y );
        }
    }
}

line( 1, 5, 2, 9 );

File: parked/line.fs
\ Draw non-orthogonal lines
\ (c)copyright 2016 by Gerald Wodni<gerald.wodni@gmail.com>

\ note: this file is a draft. I discarded the idea of non-orthogonal lines, as they would be barely used and waste a lot of code space.
\ if you are eager, feel free to finish this and send me a pull-request ;)

compiletoflash

\ : line ( x1 y1 x2 y2 -- )
\     2over 2over     ( x1 y1 x2 y2  x1 y1 x2 y2  )
\     rot -           ( x1 y1 x2 y2  x1 x2 dy     )
\     -rot swap -     ( x1 y1 x2 y2  dy dx        )
\     abs swap abs    ( x1 y2 x2 y2 |dx| |dy|     )
\     > if
\         ." x"
\     else
\         ." y"
\     then
\     ;

: ~~ .s cr ;

: clearstack depth 0 ?do drop loop ;

: line ( x1 x2 y1 y2 -- )
    ~~
    2over swap - abs        ( x1 x2 y1 y2  dx )
    ~~
    >r 2dup swap - abs r>   ( x1 x2 y1 y2  dy dx )
    ~~
    < if \ move along x
        ." x"
    else \ move along y
        ." y"
        2dup < if \ swap start/end point to move positively
            swap 2swap swap 2swap
        then
        swap 1+ swap \ increment end point
        ~~
        do
            i
            1 rpick . cr
            ~~ . cr
        loop
    then
    ~~
    clearstack
    ;
\ 1 2 5 9 line
1 5 2 9 line
